### PR TITLE
Build system improvements for pip wheel compatibility

### PR DESCRIPTION
**Pull request by tskisner**
This is still a work in progress.  So far I have:

- Simplified the cmake build system (fewer internal convenience libs)

- Improved the setup.py (which calls cmake).  This now uses mpi4py to attempt to find the MPI compiler wrappers used at install and then pass that to cmake.  This will help people pip-installing from github who also want to use MPI.

### CODE REVIEW

**Comment by tskisner** [comment]
In the spirit of having more frequent, small PRs, I would like to just merge this work as-is in terms of the build system improvements.  Actually building pip packages will be in a later PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,9 +44,24 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
...
+if(NOT TOAST_STATIC_DEPS AND ENV{TOAST_STATIC_DEPS})
+  set(TOAST_STATIC_DEPS $ENV{TOAST_STATIC_DEPS})
+endif()
+
 find_package(OpenMP)
 
-find_package(BLAS)
+if(TOAST_STATIC_DEPS)
+    set(BLA_STATIC TRUE)
+    find_package(BLAS)
```
Comment (keskitalo) [inline CMakeLists.txt:60]: Is it necessary to have `find_package(BLAS)` in both branches? Why not after the `if`?

Comment (tskisner) [reply CMakeLists.txt]: Good point, will fix that.

**Comment by tskisner** [comment]
Not sure why the travis status is not being displayed here on the PR, I just checked the build on travis-ci.org and it passed.  Merging now.

**Pull request merged**